### PR TITLE
VX_NN - vxUpsampleNearestLayer node implementation on HIP backend

### PR DESCRIPTION
**Pull request by hansely**
Implementation of vxUpsampleNearestLayer node on HIP backend which is needed for yolov3 models.

### CODE REVIEW

**Comment by hansely** [comment]
```
$ ./anntest weights.bin
OK: loaded 39 kernels from libvx_nn.so  
OK: OpenVX using GPU device#0 AMD Radeon VII (gfx906:sramecc+:xnack-) (with 60 CUs) on PCI bus 44:00.0
OK: graph initialization with annAddToGraph() took 10895.731 msec
OK: vxProcessGraph() took 1619.843 msec (1st iteration)
OK: vxProcessGraph() took 27.555 msec (average over 100 iterations)
OK: HIP buffer usage: 531069860, 376/376
OK: successful
```
This fixes #802 upsample issue.

**Comment by kiritigowda** [comment]
@hansely can you try OpenCL & HIP backend using dockers?

**Comment by hansely** [comment]
> @hansely can you try OpenCL & HIP backend using dockers?

@kiritigowda 

On OCL backend: Runs successfully with batch size 4.
On HIP backend: ./anntest weights.bin
OK: loaded 39 kernels from libvx_nn.so
"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"
Aborted (core dumped)

File: amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp [hunk 1]
```diff
@@ -156,7 +156,55 @@ static vx_status VX_CALLBACK opencl_codegen(
...
+    hipStream_t hip_stream;
+
+    ERROR_CHECK_STATUS(vxQueryNode(node, VX_NODE_ATTRIBUTE_AMD_HIP_STREAM, &hip_stream, sizeof(hip_stream)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HIP, &input_mem, sizeof(input_mem)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_OFFSET_GPU, &input_offset, sizeof(input_offset)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HIP, &output_mem, sizeof(output_mem)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_OFFSET_GPU, &output_offset, sizeof(output_offset)));
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_STRIDE_GPU, temp, sizeof(temp)));
+    input_stride.x = temp[0];
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp:188]: you can avoid this extra copy by directly passing pointer to input_stride after type casting

File: amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp [hunk 2]
```diff
@@ -156,7 +156,55 @@ static vx_status VX_CALLBACK opencl_codegen(
...
+    output_stride.w = temp[3];
+
+    if (HipExec_Upsample_Nearest_layer(hip_stream, globalThreads, dim3(1), type, input_mem, input_offset, input_stride,
+        output_mem, output_offset, output_stride)) {
+        return VX_FAILURE;
+    }
+
+    return VX_SUCCESS;
+#elif ENABLE_OPENCL
     return VX_ERROR_NOT_IMPLEMENTED;
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp:206]: OpenCL implementation was there before. We should add that back in as well

File: amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp (hunk 1 above)
Comment (hansely) [reply amd_openvx_extensions/amd_nn/src/upsample_nearest_layer.cpp:188]: Thanks Rajy for the code review. By type casting, do you mean to type cast the input_stride to vx_size? Can you elaborate little more?

**Comment by kiritigowda** [comment]
@hansely let me know when this is ready to be merged?

**Pull request merged**